src/Illuminate/Support/Number.php

```diff
@@ -367,6 +367,30 @@ public static function defaultCurrency()
         return static::$currency;
     }
 
+    /**
+     * Generate a random number of the given length.
+     *
+     * @param  int  $length
+     * @return int
+     */
+    public static function random(int $length = 6): int
+    {
+        $maxLength = strlen((string) PHP_INT_MAX);
+
+        if ($length < 1 || $length > $maxLength) {
+            return 0;
+        }
+
+        $min = 10 ** ($length - 1);
+        $max = (10 ** $length) - 1;
+
+        if ($length == $maxLength) {
```

Comment:
Even though, you are using `strlen()` here, I'd say, "digits" is the more correct term when dealing with numbers:
```suggestion
     * Generate a random number with the given amount of digits.
     *
     * @param  int  $digits
     * @return int
     */
    public static function random(int $digits = 6): int
    {
        $maxDigits = strlen((string) PHP_INT_MAX);

        if ($digits < 1 || $digits > $maxDigits) {
            return 0;
        }

        $min = 10 ** ($digits - 1);
        $max = (10 ** $digits) - 1;

        if ($digits == $maxDigits) {
```

Reply:
Yes, it is more readable.